concepts/random/about.md

```diff
@@ -0,0 +1,171 @@
+# About
+
+Many programs need (apparently) random values to simulate real-world events.
+
+Common, familiar examples include:
+- A coin toss: a random value from `('H', 'T')`.
+- The roll of a die: a random integer from 1 to 6.
+- Shuffling a deck of cards: a random ordering of a card list.
+
+Generating truly random values with a computer is a surprisingly difficult technical challenge, so you may see these results referred to as "pseudorandom".
+
+In practice, a well-designed library like the [`random`][random] module in the Python standard library is fast, flexible, and gives results that are amply good enough for most applications in modelling, simulation and games.
+
+The rest of this page will list a few of the most common functions in `random`.
+
+We encourage you to explore the full [`random`][random] documentation, as there are many more options.
+
+## Important Warning!
+
+The `random` module should __NOT__ be used for security and cryptographic applications.
+
+Instead, Python provides the [`secrets`][secrets] module.
+This is specially optimized for cryptographic security.
+
+## Create random integers
+
+The `randrange()` function has three forms, to select a random value from `range(start, stop, step)`:
+- `randrange(stop)` gives an integer `n` such that `0 <= n < stop`
+- `randrange(start, stop)` gives an integer `n` such that `start <= n < stop`
+- `randrange(start, stop, step)` gives an integer `n` such that `start <= n < stop` and `n` is in the sequence `start, start + step, start + 2*step...`
+
+For the common case where `step == 1`, the `randint(a, b)` function may be more convenient and readable.
+
+Possible results from `randint()` _include_ the upper bound, so `randint(a, b)` is the same as `randrange(a, b+1)`.
+
+```python
+>>> import random
+
+>>> random.randrange(500)
+219
+>>> [random.randrange(0, 10, 2) for _ in range(10)]
+[2, 8, 4, 0, 4, 2, 6, 6, 8, 8]
+
+>>> random.randint(1, 6)  # roll a die
+4
+```
+
+## Working with sequences
+
+The functions in this section assume that you are starting from some sequence.
+
+This will typically be a `list`, or with some limitations a `tuple` or `set` (`tuple` is immutable, and `set` is unordered).
+
+### `choice()` and `choices()`
+
+The `choice()` function will return one random entry from a sequence.
+
+At its simplest, the coin-flip example:
+
+```python
+>>> [random.choice(['H', 'T']) for _ in range(5)]
+['T', 'H', 'H', 'T', 'H']
+```

```

Comment:
```suggestion
```python
# This will pick one of the two values in the list at random 5 separate times 
>>> [random.choice(['H', 'T']) for _ in range(5)]
['T', 'H', 'H', 'T', 'H']
```

```